src/util/color/colorpalette.cpp

```diff
@@ -5,6 +5,17 @@ mixxx::RgbColor ColorPalette::nextColor(mixxx::RgbColor color) const {
     return at((indexOf(color) + 1) % size());
 }
 
+mixxx::RgbColor::optional_t ColorPalette::nextColor(mixxx::RgbColor::optional_t color) const {
+    if (color) {
+        // If color is the last element in the palette, return no color
```

Comment:
Should this wrap around to the first color instead of returning nullopt?

Reply:
No, we already have a `nextColor` for non-optional colors. This is specifically for optional colors like track color.

Reply:
Okay I understand now. This effectively includes nullopt in the cycle of the palette.